### PR TITLE
feat: redesign teleport bridge

### LINKED ISSUE #4119: Revamp XCM Teleport + Assets
Right now, first-time users, especially that outside of Polkadot have no idea what Basilisk is, or what Ksm is at least, and they have way lot of issues getting assets on the chain to buy/mint/interact at least. To ease this good first step would be to implement XCM teleporting of KSM to Basilisk, so users can skip context switching going to Karura bridge and selecting stuff. 
The goal is to mimic the same functionality with our dialogues, so users have an easier job doing this stuff. 

Plus, I'm thinking of merging it with the assets screen so that users can teleport right away from that screen and see results as XCM transfers are quite fast. 
This will open at least some one dialog to supporting multi-currencies and showing on various address, hopefully we will make it right so it's reliable for other NFT artists out there

Will go with @exezbcz to speedrun this design so we have something quickly to implement and drop here once things will be on stage :) 

#### Suggestions 
from Luu to have visual showed KSM & BSX chain and others probably

---
## Design

Desktop - 14 (1)
design is available now in Figma - gallery item handoff - https://www.figma.com/file/gHSzqGPeQ9qK4L0qIDcTjY/gallery-item-handoff?node-id=1270%3A5615
How it works: 
### Buttons:
- please see #4111 section with the buttons
- send button has same hover states as primary in the gallery detail. 
- other buttons behaves similarly to button secondary 
- blue links don't have a hover effect, only a cursor pointer. 

### Description
- top part:
    - heading and your address, buttons for logout and copy adress
- teleport: 
    - network you want to bridge from - selection, scalable in need of more networks
    - input field, Right side of the card isn't clickable and there is no hover effect, currency is fixed for now. In future it would be changed and it would trigger a dropdown with more options. 
    - Destination, same as "from"
    - dialog: You will receive “amount” on “network” to "address", which functions as a confirmation of the step you want to make. 
    - receive funds to another address
    - send button, behaves as a primary button - hover effect - white bg
- Assets
   - some text
   - table with assets, network and balance
- Your Fee asset
   -  explaining text and then fee asset selection - dropdown with asset options
image

---
Old teleport on RMRK to teleport KSM to Statemine
image

## Ref
- #4035 
- #4036 
- #4118 
- https://apps.karura.network/bridge
- https://beta.kodadot.xyz/bsx/assets
- https://wiki.polkadot.network/docs/learn-teleport
- https://youtu.be/B5EvysihBFc?t=155

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

## Context

- [x] Closes #4119
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by vikiival** [comment]
Address you need to encode and decode the address (check transfer (should be like original address)).

[The thing is that addresses are encoded by ss58](https://polkadot.js.org/docs/keyring/start/ss58)

First you need to decode address via `decodeAddress`
then you need to encode it with bsx ss58 (should be saved in vuex imo)

TL;DR 

we have utilities for that in `account.ts`

**Comment by Jarsen136** [comment]
> Address you need to encode and decode the address (check transfer (should be like original address)).
> 
> 
> 
> [The thing is that addresses are encoded by ss58](https://polkadot.js.org/docs/keyring/start/ss58)
> 
> 
> 
> First you need to decode address via `decodeAddress`
> 
> then you need to encode it with bsx ss58 (should be saved in vuex imo)
> 
> 
> 
> TL;DR 
> 
> 
> 
> we have utilities for that in `account.ts`
> 
> 
> 
> 

Thanks a lot. It really solves my problem 👍

**Comment by Jarsen136** [comment]
TODO:
- [ ] show balance on current address account
- [ ] calculate the usd value of KSM to send

**Comment by yangwao** [comment]
> TODO:

follow up issue? 

I guess we can meanwhile use it into videos :)

**Comment by Jarsen136** [comment]
> > TODO:
> 
> follow up issue?
> 

I'm working on it.

**Comment by Jarsen136** [comment]
> TODO:
> 
> * [ ]  show balance on current address account
> * [ ]  calculate the usd value of KSM to send
> * [ ]  link Subscan to Basilisk address


✅  show balance on basilisk  address account
✅  calculate the usd value of KSM to send
✅  link Subscan to Basilisk address /  Kusama address

image



make follow-up issue:
 - [ ] show balance on Kusama address
image

**Comment by yangwao** [comment]
let's see what @exezbcz thinks 🤔

**Comment by exezbcz** [comment]
yes yes, let me check :D 
## Small things
image
- button on hover: the text changes (it should stay black) 

image
- link color to k-blue

image
- input field to k-dark in darkmode

image
- capitalize "s" please
 ## Disabled 
lets add disabled button when you input 0 or amount you dont have - 
styles: 
### nothing inputed
image
- also when you input zero. 

### Not enough balance
image- - 

### All ok
image

- I was also thinking about some confirmation dialog, but that maybe in v2 
## Balance indicator 
image
- it shows bsx instead of ksm (i dont have bsx on my wallet
image
- when bridging from ksm it does not show the balance but it was already mentioned. 


Good job!

**Comment by Jarsen136** [comment]
All the issues mentioned above have been solved, except for these two. I have made an issue for them #4913
Pls take a look. @exezbcz 

> when bridging from ksm it does not show the balance but it was already mentioned.

> Not enough balance
image

**Comment by exezbcz** [comment]
looks good! 
one thing is the dark input
image
- text has black color - no contrast

image
- max link should have the same shade of blue like address link :D

**Comment by Jarsen136** [comment]
> looks good! one thing is the dark input image
> 
> * text has black color - no contrast
> 
> image
> 
> * max link should have the same shade of blue like address link :D

✅ Done。 Thanks for pointing that out.

**Comment by exezbcz** [comment]
> > looks good! one thing is the dark input image
> > 
> > * text has black color - no contrast
> > 
> > image
> > 
> > * max link should have the same shade of blue like address link :D
> 
> ✅ Done。 Thanks for pointing that out.

yes!

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 36.59 USD/KSM ~ 1.366 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x5465b5d11588e54685d6f4810f96998f9c50b444bd6161a6b629ccf0f5997ae9](https://kusama.subscan.io/extrinsic/0x5465b5d11588e54685d6f4810f96998f9c50b444bd6161a6b629ccf0f5997ae9)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)
